src/boot/AppEventHandlers.js

```diff
@@ -121,9 +123,33 @@ class AppEventHandlers extends PureComponent<Props> {
     // Release memory here
   };
 
-  componentDidMount() {
+  /**
+   * Do one-off tasks that require persisted data to have loaded.
```

Comment:
> Hmm dang -- did you observe this crash empirically?

I did observe a crash while investigating my suspicion of a crash, but I'm not reproducing it right now; I'll try more tomorrow?

> The existing strategy we have that should prevent this kind of thing is, we use `redux-action-buffer` to cause all actions to be held up in a queue until a REHYDRATE action comes along.

True, but I was specifically thinking to prepare for the future when navigation actions aren't dispatched through Redux and `redux-action-buffer` will have no power over them. But you're right; now I think more closely about it, being in a thunk action creator should do the trick (and it seems to in my testing just now).

Reply:
> > Hmm dang -- did you observe this crash empirically?
> 
> I did observe a crash while investigating my suspicion of a crash, but I'm not reproducing it right now; I'll try more tomorrow?

OK, I think I've solved this mystery—I was logged out when I observed the crash. If I set it up so that `handleInitialNotification` dispatches `doNarrow` unconditionally (I had it do so before `await getInitialNotification()`), it *does* crash if the user is logged out; I get the "no server data" message. No real surprise there.

But this case isn't representative—`narrowToNotification` will handle checking to see that the account the notification is meant for is logged in.

Reply:
Great -- glad that's diagnosed.